### ISSUE TITLE
Support different levels of confidence estimates when using mokapot

### ISSUE THREAD

**Issue by jonasscheid** [compomics/ms2rescore#112]
Currently PSMs are evaluated using psm-level FDR when using mokapot [see here](https://github.com/jonasscheid/ms2rescore/blob/fea1df1ff2bca68988136150da9809b3a32b3b06/ms2rescore/rescoring_engines/mokapot.py#L96)
Peptide-level confidence estimates are also [present](https://mokapot.readthedocs.io/en/latest/api/confidence.html#mokapot.confidence.LinearConfidence) in the object mokapot returns. Would be of interest to give the user the freedom to choose the FDR-level

**Comment by RalfG** [comment]
Agreed that this could be reported somehow. The main issue blocking this is that the (output) object in MS²Rescore is the PSM list, which has only PSM-level entries. Would you repeat each peptide/protein-level q-value and PEP for each PSM linked to these? Remaining issue then is how to deal with PSMs linked to multiple protein groups. Or am I missing something? Ideas are very much welcome!

**Comment by jonasscheid** [comment]
> Would you repeat each peptide/protein-level q-value and PEP for each PSM linked to these?
Yes I had that also in mind! But for peptides mapping to multiple proteins that is indeed a tricky one..
One ad-hoc idea I have for protein-level fdr is to store a dict of proteins with the confidence estimates in `provenance_data`, but then again `provenance_data : dict[str, str]` so that is currently not possible. Maybe only support peptide-level-fdr for now? 🤔
